pkg/operator/csidriveroperator/deploymentcontroller.go

```diff
@@ -0,0 +1,207 @@
+	} else {
+		deploymentAvailable.Status = operatorapi.ConditionFalse
+		deploymentAvailable.Reason = "WaitDeployment"
+		deploymentAvailable.Message = "Waiting for a Deployment pod to start"
+	}
+
+	// Not progressing: all replicas are at the latest version && Deployment generation matches
+	deploymentProgressing := operatorapi.OperatorCondition{
+		Type: c.name + deploymentControllerName + operatorapi.OperatorStatusTypeProgressing,
+	}
+	if deployment.Status.ObservedGeneration != deployment.Generation {
+		deploymentProgressing.Status = operatorapi.ConditionTrue
+		deploymentProgressing.Reason = "NewGeneration"
+		msg := fmt.Sprintf("desired generation %d, current generation %d", deployment.Generation, deployment.Status.ObservedGeneration)
+		deploymentProgressing.Message = msg
+	} else {
+		if deployment.Spec.Replicas != nil {
+			if deployment.Status.UpdatedReplicas == *deployment.Spec.Replicas {
+				deploymentProgressing.Status = operatorapi.ConditionFalse
+				// All replicas were updated, set the version
+				c.versionGetter.SetVersion(c.name+versionName, c.targetVersion)
+			} else {
+				msg := fmt.Sprintf("%d out of %d pods running", deployment.Status.UpdatedReplicas, *deployment.Spec.Replicas)
+				deploymentProgressing.Status = operatorapi.ConditionTrue
+				deploymentProgressing.Reason = "WaitDeployment"
+				deploymentProgressing.Message = msg
+			}
+		}
+	}
+
+	resourcemerge.SetDeploymentGeneration(&opStatus.Generations, deployment)
+	opStatus.ReadyReplicas = deployment.Status.ReadyReplicas
+
+	updateGenerationFn := func(newStatus *operatorapi.OperatorStatus) error {
+		if deployment != nil {
+			resourcemerge.SetDeploymentGeneration(&newStatus.Generations, deployment)
+		}
+		return nil
+	}
+
+	if _, _, err := v1helpers.UpdateStatus(c.operatorClient,
+		v1helpers.UpdateConditionFn(deploymentAvailable),
+		v1helpers.UpdateConditionFn(deploymentProgressing),
+		updateGenerationFn,
+	); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *CSIDriverOperatorDeploymentController) getRequiredDeployment(spec *operatorapi.OperatorSpec) *appsv1.Deployment {
+	deploymentAsset := c.csiOperatorConfig.DeploymentAsset
+	deploymentString := string(generated.MustAsset(deploymentAsset))
+
+	// Replace images
+	if c.csiOperatorConfig.ImageReplacer != nil {
+		deploymentString = c.csiOperatorConfig.ImageReplacer.Replace(deploymentString)
+	}
+	deploymentString = sidecarReplacer.Replace(deploymentString)
+
+	// Replace log level
+	logLevel := getLogLevel(spec.LogLevel)
+	deploymentString = strings.ReplaceAll(deploymentString, "${LOG_LEVEL}", strconv.Itoa(logLevel))
+
+	deployment := resourceread.ReadDeploymentV1OrDie([]byte(deploymentString))
+	return deployment
+}
+
+func (c *CSIDriverOperatorDeploymentController) Run(ctx context.Context, workers int) {
+	// This adds event handlers to informers.
+	ctrl := c.factory.WithSync(c.Sync).ToController(c.name+csiDriverControllerName, c.eventRecorder)
```

Comment:
did you mean deploymentControllerName instead of csiDriverControllerName?

or use c.Name() instead?

Reply:
It's recommended to prefix the event recorder as well: `eventRecorder.WithComponentSuffix()`

Reply:
Added suffix to all new controllers. Using "short" name for the CSI driver ones.